src/sidebar/components/test/user-menu-test.js

```diff
@@ -0,0 +1,248 @@
+'use strict';
+
+const { createElement } = require('preact');
+const { shallow } = require('enzyme');
+
+const UserMenu = require('../user-menu');
+const MenuItem = require('../menu-item');
+
+describe('UserMenu', () => {
+  let fakeAuth,
+    fakeBridge,
+    fakeIsThirdPartyUser,
+    fakeOnLogout,
+    fakeProfileBridgeEvent,
+    fakeServiceConfig,
+    fakeServiceUrl,
+    fakeSettings;
+
+  const createUserMenu = () => {
+    return shallow(
+      <UserMenu
+        auth={fakeAuth}
+        bridge={fakeBridge}
+        onLogout={fakeOnLogout}
+        serviceUrl={fakeServiceUrl}
+        settings={fakeSettings}
+      />
+    ).dive(); // Dive needed because this component uses `withServices`
+  };
+
+  const findMenuItem = (wrapper, labelText) => {
+    return wrapper
+      .find(MenuItem)
+      .filterWhere(n => n.prop('label') === labelText);
+  };
+
+  beforeEach(() => {
+    fakeAuth = {
```

Comment:
In future I'd like to get rid of the `vm.auth` property in favour of just looking up the user profile in the store, as per @hmstepanek's ongoing refactorings. This is fine for now though.